### PR TITLE
plugin: add `keys` support for `Memento`

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request adds support for `keys()` in the `Memento` VS Code API.

#### How to test

1. include the following test plugin ([vscode-memento-keys-0.0.1.vsix.zip](https://github.com/eclipse-theia/theia/files/9221698/vscode-memento-keys-0.0.1.vsix.zip)).
2. start the application with a workspace.
3. execute the command `Memento: Workspace Keys` - a notification should appear with workspace keys (denoted by `w-` prefix.
4. execute the command `Memento: Global Keys` - a notification should appear with global keys (workspace keys should also appear).

The result should be the same as in VS Code:

Screen Shot 2022-07-29 at 12 33 19 PM

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/plugin-ext/src/plugin/plugin-storage.ts [hunk 1]
```diff
@@ -38,6 +38,10 @@ export class Memento implements theia.Memento {
         }
     }
 
+    keys(): string[] {
+        return Object.entries(this.cache ?? {}).filter(([, value]) => value !== undefined).map(([key]) => key);
```
Comment (msujew) [inline packages/plugin-ext/src/plugin/plugin-storage.ts:42]: Question: I'm not too familiar with this code, but it seems like `this.cache` can never be undefined, right? Or are there cases in which the `?? {}` actually does something? If so, we should probably align `get<T>` and `update` so they also correctly deal with `cache` being undefined.

**Review by msujew — COMMENTED** [review]
The change looks good to me. I have a question though, see below.

File: packages/plugin-ext/src/plugin/plugin-storage.ts (hunk 1 above)
Comment (vince-fugnitto) [reply packages/plugin-ext/src/plugin/plugin-storage.ts]: You're right, we always make sure to properly initialize the cache so it is never `undefined`. 
I've updated the pull-request.

**Review by msujew — APPROVED** [review]
LGTM 👍

**Pull request merged**